source/Calamari/Kubernetes/Integration/HelmCli.cs

```diff
@@ -102,6 +112,12 @@ public CommandResult Upgrade(string releaseName, string packagePath, IEnumerable
             buildArgs.Add(releaseName);
             buildArgs.Add(packagePath);
 
+            if (OctopusFeatureToggles.ExecuteHelmUpgradeCommandViaShellScriptFeatureToggle.IsEnabled(variables))
+            {
+                log.Warn("Backwards compatibility for Helm command execution via bash scripts is temporary and will be removed in a future release. Provide arguments compatible with direct Helm invocation, avoiding bash-specific formatting.");
```

Comment:
Usually I like to give some indication of when a feature might be removed. Since I'd argue this isn't a standard deprecation, but a bugfix that the customer was just relying on, we probably don't need a full 12 month notice.

Lets aim for `2025.3` (6 months) before we strip it out. We can always take longer if we need, but this gives us a target timeframe that, if we hear no (good) objections we can actually take the next step.

When we have a public issue that links to this change its probably worth including a reference to this as well.